app/src/main/java/com/kennedy/peter/bloquery/firebase/User.java

```diff
@@ -0,0 +1,15 @@
+package com.kennedy.peter.bloquery.firebase;
+
+public class User {
+    public String UID, userName, email;
+    public void updateUserDetails(String UID, String userName, String email) {
+        this.UID = (UID == null) ? this.UID : UID;
```

Comment:
This is very confusing. What would be the scenario in which you'd update with `null`s ?